### ISSUE TITLE
Support alt text property via the PasteImageDialog

### ISSUE THREAD

**Issue by martingreilinger** [asciidoctor/asciidoctor-intellij-plugin#566]
## Overview:
As previously suggested in [PR #559](https://github.com/asciidoctor/asciidoctor-intellij-plugin/pull/559#issuecomment-703077250), it would be a good idea to support the alt text property via the `PasteImageDialog`

## Technical Description:
Functionality would be similar to the _include width_ field.
The dialog should consist of:
* Check box which includes or excludes the field (enables/disables the text field when checked/unchecked)
* Text field (I suggest using the `JBTextField` for its "native" feeling)

Text field validation:
* I suggest only allowing letters as this should be a descriptive text after all

During dialog submit:
* Return an `Optional<String>` which is
  * `empty` if the check box is unchecked or the field is empty
  * *contains value* if the checkbox is checked an the field is not empty

Suggested layout:
![suggested-layout](https://user-images.githubusercontent.com/25162092/95785860-d457f180-0cd6-11eb-9b28-4deea4c66d9e.png)
(This example uses a `GridBagLayout` to increase the text field width. Additional y-padding was added to match the width input.)
**NOTE:** The input fields could benefit from a reduced y-padding and the width input could be slimmer in my opinion.
If this opinion is shared, I will create a ticket to align the width field fist.

Output:
* If a value is present it should be added to the other image arguments
  * The value should be assigned to the `alt` label (see [image attributes](https://asciidoctor.org/docs/user-manual/#summary-2))
  * The value should be quoted
  **NOTE:** It appears this does not matter based on my testing, but other attributes suggest this like `title` or `caption` so to stay consistent, I figured it would not hurt

### Acceptance Criteria:

```gherkin
Feature: Support alt text property via the PasteImageDialog

  Background:
    Given an image is pasted into an Asciidoc file

  Scenario: Alt text is included
    Given the "include alt text" check box is selected
    And the alt text "Mountain range covered in clouds" is entered
    When the dialog is submitted
    Then the created image macro should contain an "alt" attribute
    And the value should be "Mountain range covered in clouds"
```

* Include alt text: `"Mountainrange covered in clouds"`
* Output: `image::<img-reference>[alt="Mountainrange covered in clouds"]`

I am happy to discuss anything before I start with the implementation.
_NOTE: We can also discuss the issue structure, in regards to #564_

**Comment by ahus1** [comment]
Hi @martingreilinger - thank you for the detailed outline for this planned change. 

If you think the existing existing fields of this dialog need refactoring or rework (like the reduced y-padding), please do this part of this issue as part of the "scout's rule". Don't open a new issue for that as it would consider it too much overhead.

I'm also OK if you choose to add other attributes to the dialog, at least the ones you consider common fields. Please be selective: There are a lot of fields for images, adding all of them would make the dialog unusable.

I agree that the value should be quoted when inserting it to the source. I would consider this a best practice.   

For input validation you suggested "only allowing letters". I think this is too restrictive; people would want to use punctuation, numbers and even quotes. Please allow all characters, including quotes. Adding newlines shouldn't be possible in the field AFAIK, so no validation necessary here. A double quote should be escaped (`"` -> `\"`) to have a correct AsciiDoc syntax before inserting it into the source.

Unfortunately the HTML backend of Asciidoctor will not escape double quotes when creating HTML what is a pity. This shouldn't be a concern here. It should be handled upstream.

Regarding the issue structure: The outline you used for this issue captures the aspects of design and test. The existing templates capture aspects of bugs and missing features (issue template used by users of the plugin) or a new contribution (pull request template used by contributors). 
I therefore suggest to keep the aspects of design and test separate; maybe keep it in the contributor's guide as part of a step-by-step guide for contributors.

**Comment by martingreilinger** [comment]
@ahus1 Thanks for answering my questions, sounds good!

**Comment by martingreilinger** [comment]
### Update
After consolidating the [Intellij UI Guidlines](https://jetbrains.design/intellij/), I have decided to rework the attribute section of the dialog. 

This is the new, updated layout:
![layout-overhaul](https://user-images.githubusercontent.com/25162092/96298318-8c94db00-0ff2-11eb-81b4-7e77a54aa76a.png)

**Comment by ahus1** [comment]
Thank you for looking into the UI Guidelines. The new layout looks way better.  🥇

I have to admit that I never consulted the UI Guidelines before and only tried to make my dialogues similar to those I knew. I assume that you now know more about them than I do.

**Comment by ahus1** [comment]
This now available as part or pre-release 0.31.35. Thank you for making this dialog shine!

The pre-release of the plugin is available from [GitHub releases](https://github.com/asciidoctor/asciidoctor-intellij-plugin/releases) and the [IntelliJ AsciiDoc EAP repository](https://intellij-asciidoc-plugin.ahus1.de/docs/contributors-guide/user/add-eap-repository-to-ide.html).
